TSPL.docc/ReferenceManual/Types.md

```diff
@@ -347,10 +347,43 @@ that is, a function that takes an `Int` and returns
 another function that takes and returns an `Int`.
 
 Function types for functions
-that can throw or rethrow an error must be marked with the `throws` keyword.
-The `throws` keyword is part of a function's type,
-and nonthrowing functions are subtypes of throwing functions.
-As a result, you can use a nonthrowing function in the same places as a throwing one.
+that can throw or rethrow an error must include the `throws` keyword.
+<!-- XXX TR: Confirm rethrowing functions use 'throws' -->
+You can include a type after `throws` in parentheses
+to specify the type of error that the function throws.
+The throw error type must conform to the `Error` protocol.
+Writing `throws` without specifying an type
+is the same as writing `throws(any Error)`.
+Omitting `throws` is the same as writing `throws(Never)`.
+The error type that a function throws
+can be any type that conforms to `Error`,
+including generic types, boxed protocol types, and opaque types.
+
+The type of error that a function throws is part of that function's type,
+and a subtype relationship between error types
+means the corresponding function types are also subtypes.
+For example, if you declare a custom `MyError` type,
+the relationship between some function types is as follows,
+from supertype to subtype:
+
+1. Functions that throw any error, marked `throws(any Error)`.
+1. Functions that throw a specific error, marked `throws(MyError)`.
+1. Functions that don't throw, marked `throws(Never)`.
```

Comment:
I wouldn't include periods because these aren't complete sentences, unless it is your style in TSPL to put periods at the end of all bulleted lists.
```suggestion
1. Functions that throw any error, marked `throws(any Error)`
1. Functions that throw a specific error, marked `throws(MyError)`
1. Functions that don't throw, marked `throws(Never)`
```